modules/tinymce/src/core/test/ts/browser/keyboard/EnterKeyTest.ts

```diff
@@ -681,4 +682,21 @@ describe('browser.tinymce.core.keyboard.EnterKeyTest', () => {
       TinyAssertions.assertContent(editor, initialContent);
     });
   });
+
+  it('TINY-11753: pressing enter when the seleciton is in a CEF element should move the selection into the first CET element', () => {
+    const editor = hook.editor();
+    editor.setContent('<figure contenteditable="false"><img src="file.png"><figcaption contenteditable="true">Caption</figcaption></figure>');
+    TinySelections.select(editor, 'figure', []);
+    TinyAssertions.assertSelection(editor, [], 0, [], 1);
+
+    TinyContentActions.keystroke(editor, Keys.enter());
+    TinyAssertions.assertCursor(editor, [ 0, 1, 0 ], 0);
+
+    editor.setContent('<div contenteditable="false"><img src="file.png"><span contenteditable="true">Caption</span></div>');
```

Comment:
Could we split this up into two separate `it` cases. Since then you can describe each case in the test label.

Reply:
done